libgadget/cooling.c

```diff
@@ -67,44 +69,42 @@ double DoCooling(double redshift, double u_old, double rho, double dt, struct UV
     LambdaNet = get_heatingcooling_rate(rho, u, 1 - HYDROGEN_MASSFRAC, redshift, Z, uvbg, ne_guess);
 
     /* bracketing */
-
-    if(u - u_old - LambdaNet * dt < 0)	/* heating */
+    if(LambdaNet * dt > 0)	/* heating */
```

Comment:
No predict any more?

Reply:
We have u = u_old above this so for this initial step the removed code did nothing. I can put it back though if you like.